src/OrleansProviders/Streams/Common/Monitors/DefaultBlockPoolMonitor.cs

```diff
@@ -0,0 +1,52 @@
+﻿using Orleans.Runtime;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Orleans.Providers.Streams.Common
+{
+    /// <summary>
```

Comment:
These default classes seem to be intended to be overridden, do you foresee that being common?
I'd avoid the virtual functions, unless we really think that will be a common pattern.